shared/git/row/index.js

```diff
@@ -0,0 +1,303 @@
+
+import {globalStyles, globalColors, globalMargins, transition} from '../../styles'
+import {isMobile} from '../../constants/platform'
+
+export type Props = {
+  canDelete: boolean,
+  devicename: string,
+  expanded: boolean,
+  lastEditTime: string,
+  lastEditUser: string,
+  lastEditUserFollowing: boolean,
+  name: string,
+  teamname: ?string,
+  url: string,
+  isNew: boolean,
+  onCopy: () => void,
+  onShowDelete: () => void,
+  onToggleExpand: () => void,
+  setTimeout: (() => void, number) => number,
+}
+
+type State = {
+  showingCopy: boolean,
+}
+
+class Row extends React.Component<Props, State> {
+  state = {
+    showingCopy: false,
+  }
+
+  _input: any
+
+  _inputOnClick = () => {
+    this._input && this._input.select()
+  }
+
+  _setRef = r => (this._input = r)
+
+  _onCopy = () => {
+    this.props.onCopy()
+    this.setState({showingCopy: true})
+    this.props.setTimeout(() => this.setState({showingCopy: false}), 1000)
+  }
+
+  render() {
+    return (
+      <Box
+        style={{
+          ..._rowStyle,
+          ...(this.props.expanded
+            ? {
+                backgroundColor: globalColors.blue5,
+                borderColor: globalColors.black_05,
+              }
+            : {}),
+        }}
+      >
+        <ClickableBox
+          onClick={this.props.onToggleExpand}
+          style={_rowClickStyle}
+          hoverColor={isMobile ? undefined : globalColors.transparent}
+          underlayColor={globalColors.transparent}
+        >
+          <Box style={_rowTopStyle}>
+            <Icon
+              type={this.props.expanded ? 'iconfont-caret-down' : 'iconfont-caret-right'}
+              style={_iconCaretStyle}
+            />
+            <Icon
+              type={this.props.teamname ? 'iconfont-repo-team' : 'iconfont-repo-personal'}
+              style={_iconRepoStyle}
+            />
+            {this.props.teamname &&
+              <Avatar
+                size={12}
+                isTeam={true}
+                teamname={this.props.teamname}
+                style={{marginRight: globalMargins.xtiny}}
+              />}
+            <Text type="BodySemibold" style={{color: globalColors.darkBlue}}>
+              {this.props.teamname ? `${this.props.teamname}/${this.props.name}` : this.props.name}
+            </Text>
+            {!!this.props.teamname && this.props.isNew && <Meta title="New" style={_metaStyle} />}
+          </Box>
+        </ClickableBox>
+        {this.props.expanded &&
+          <Box style={_rowBottomStyle}>
+            <Box
+              style={{
+                ...globalStyles.flexBoxRow,
+                alignItems: 'center',
+                flexWrap: 'wrap',
+                marginBottom: 5,
+                marginTop: 2,
+              }}
+            >
+              <Text type="BodySmall">
+                Last push
+                {' '}
+                {this.props.lastEditTime}
+                {' '}
+                ago
```

Comment:
it looks like "ago" is actually included in the lastEditTime already, so it gets doubled

Reply:
(and if it's possible to get rid of the ugly `{' '}` out somehow, would be cool)

Reply:
that comes from prettier

Reply:
Yeah, I know it comes from prettier, wondering if there's something like a diff string syntax to use that stops prettier from doing that.  Not a big deal.